File: Go/2582_Pass_the_Pillow/solution.go
package solution

func passThePillow(n int, time int) int {
	cycle := time / (n - 1)
	complement := time % (n - 1)

	if cycle%2 == 1 {
		return n - complement
	} else {
		return 1 + complement
	}
}

File: JavaScript/2582_Pass_the_Pillow/solution.js
/**
 * @param {number} n
 * @param {number} time
 * @return {number}
 */
const passThePillow = function (n, time) {
    const cycle = Math.floor(time / (n - 1));
    const complement = (time % (n - 1));

    switch (cycle % 2) {
        case 0:
            return (1 + complement);
        case 1:
            return (n - complement);
    }
};